src/java/org/apache/cassandra/db/commitlog/CommitLogDescriptor.java

```diff
@@ -220,7 +223,20 @@ public String fileName()
 
     public String cdcIndexFileName()
     {
-        return FILENAME_PREFIX + version + SEPARATOR + id + "_cdc.idx";
+        return FILENAME_PREFIX + version + SEPARATOR + id + INDEX_FILENAME_SUFFIX;
+    }
+
+    /**
+     * Infer the corresponding cdc index file using its cdc commitlog file
+     * @param cdcFile
+     * @return cdc index file
+     */
+    public static File inferCdcIndexFile(File cdcFile)
+    {
+        Preconditions.checkArgument(isValid(cdcFile.name()), "Invalid commit log");
```

Comment:
It looks like this can throw an exception up stack and leave us with an inability to allocate new commit log files, repeatedly failing if the oldest CommitLogSegment is invalid.

Reply:
The check should not throw for the current usage in the codebase. The parameter has already been verified at the caller. 
Since the method is to infer, I think it would make sense to return `null` when failing to guess.